### ISSUE TITLE
storaged: boot keeps looping forever when cache disk failed to be mounted again

### ISSUE THREAD

**Issue by zaibon** [threefoldtech/zos#472]
```
[+] storaged: 2020-01-08T16:25:30Z info Initializing storage module
[+] storaged: 2020-01-08T16:25:30Z info Remounting existing volumes
[+] storaged: 2020-01-08T16:25:30Z debug Searching for existing volumes
[+] storaged/test: /var/cache is not a mountpoint
[+] storaged: 2020-01-08T16:25:31Z debug Volume 35ac6db2-96e3-471c-ae35-e6bb5a180507 already mounted
[+] storaged: 2020-01-08T16:25:31Z info Finding free disks
[+] storaged: 2020-01-08T16:25:31Z info Creating new volumes using policy single
[+] storaged: 2020-01-08T16:25:31Z debug Creating 0 new volumes
[+] storaged: 2020-01-08T16:25:31Z debug Creating 0 new volumes
[+] storaged: 2020-01-08T16:25:31Z info Making sure new volumes are mounted
[+] storaged: 2020-01-08T16:25:31Z info Setting up cache
[+] storaged: 2020-01-08T16:25:31Z debug Checking pools for existing cache
[+] storaged: 2020-01-08T16:25:31Z debug No cache found, try to create new cache
[+] storaged: 2020-01-08T16:25:31Z info start storage pool maintained pool=35ac6db2-96e3-471c-ae35-e6bb5a180507
[+] storaged: 2020-01-08T16:25:31Z info finished storage pool maintained pool=35ac6db2-96e3-471c-ae35-e6bb5a180507
[+] storaged: 2020-01-08T16:25:31Z fatal Error initializing storage module: Not enough space left in pools of this type SSD
```

**Comment by maximevanhees** [comment]
What actions should we need to take when this happens?
- Should we try to repair the cache disk?
- Should we create a new cache disk? But, then the Node ID is lost
- What would happen if we try to create a new cache disk, but there is no space available on the node?

**Comment by maximevanhees** [comment]
Reproduced the bug by corrupting the current cache (added random bytes to the cache device). Storaged will try to make new cache, and keeps looping because:
- There is no space left on any SSD
- There is no HDD available

So basically there is no place at all where the cache disk can be mounted. The scenario above is pretty exotic (i think) but should we bother to find a solution to it? If so, how could we find free space to mount the cache disk?

**Comment by muhamadazmy** [comment]
@zaibon We need your input for this. I actually can't see a clean path out of this issue. The only solution i can think of is:
- Try to clear space, but storaged can't decide if a volume can be deleted or not, that's provisiond job, but it can't start because storaged can't start.
- The other "half" solution is create a tmpfs cache in memory, but while this will allow daemons to start, there is no persistance. but may be we can then trigger a notification or something (push something to the explorer) that the node needs attention, and may be stop provisioning from actually deploying workload. Basically run in limited mode where u can (hopefully) clear up some storage

what do u think ?

**Comment by zaibon** [comment]
> Try to clear space, but storaged can't decide if a volume can be deleted or not, that's provisiond job, but it can't start because storaged can't start.

Yep this is a dead end, we cannot know what to get rid of or not.

The second proposal seems the way to go. We at some point discussed this kind of flow where we would send some notification to the farm to says something is wrong.
This indeed requires to be able to pause provisiond so we `freeze` the machine.
We can see where we end with this kind of solution

**Comment by zaibon** [comment]
fixed in #565
